src/axom/klee/Shape.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/klee/Shape.hpp"
+
+#include <algorithm>
+#include <stdexcept>
+#include <utility>
+
+namespace axom
+{
+namespace klee
+{
+namespace
+{
+/**
+ * Check whether a given container contains a value
+ *
+ * \tparam Container the type of the container
+ * \param container the container to check for the value
+ * \param value the value to look for
+ * \return whether the container contains the value
+ */
+template <typename Container>
+bool contains(const Container &container,
+              const typename Container::value_type &value)
+{
+  using std::begin;
+  using std::end;
+  auto endIter = end(container);
+  return std::find(begin(container), endIter, value) != endIter;
+}
+}  // unnamed namespace
+
+Shape::Shape(std::string name,
+             std::string material,
+             std::vector<std::string> materialsReplaced,
+             std::vector<std::string> materialsNotReplaced,
+             Geometry geometry)
+  : m_name(std::move(name))
+  , m_material(std::move(material))
+  , m_materialsReplaced(std::move(materialsReplaced))
+  , m_materialsNotReplaced(std::move(materialsNotReplaced))
+  , m_geometry(std::move(geometry))
+{
+  if(!m_materialsNotReplaced.empty() && !m_materialsReplaced.empty())
+  {
+    throw std::logic_error(
```

Comment:
Would we want to extend `KleeError` for cases like these?

Reply:
We throw a KleeError in IOError.cpp (line 159) when validating input. KleeError is used to report input errors by the user. This instance is a logic error by the programmer.